src/WebJobs.Script/Utility.cs

```diff
@@ -569,23 +569,23 @@ internal static bool IsSingleLanguage(IEnumerable<FunctionMetadata> functions, s
             {
                 throw new ArgumentNullException(nameof(functions));
             }
-            var functionsListWithoutProxies = functions.Where(f => f.IsProxy == false).ToArray();
-            if (functionsListWithoutProxies.Length == 0)
+            var functionsListWithoutProxiesCodeless = functions.Where(f => !f.IsCodeless).ToArray();
+            if (functionsListWithoutProxiesCodeless.Length == 0)
             {
                 return true;
             }
             if (string.IsNullOrEmpty(workerRuntime))
             {
-                return functionsListWithoutProxies.Select(f => f.Language).Distinct().Count() <= 1;
+                return functionsListWithoutProxiesCodeless.Select(f => f.Language).Distinct().Count() <= 1;
             }
-            return ContainsFunctionWithWorkerRuntime(functionsListWithoutProxies, workerRuntime);
+            return ContainsFunctionWithWorkerRuntime(functionsListWithoutProxiesCodeless, workerRuntime);
         }
 
         internal static string GetWorkerRuntime(IEnumerable<FunctionMetadata> functions)
         {
             if (IsSingleLanguage(functions, null))
             {
-                var functionsListWithoutProxies = functions?.Where(f => f.IsProxy == false);
+                var functionsListWithoutProxies = functions?.Where(f => !(f is ProxyFunctionMetadata));
```

Comment:
Rename. Also why do you filter out all codeless in the IsSingleLanguage call, but here only filter Proxies? Shouldn't the same filter be used both places? If so, move the filter above the IsSingleLanguage call and pass the filtered list in.

I'm not sure anyone is going to be able to understand this, and even you/I won't be able to a week from now :)

Reply:
I think filtering all codeless will be fine here. It's not as relevant.
I didn't really touch this code path, except change how we are checking for if something is proxy.

But yeah, you have a point and I can update this as well.